Codec/DicomJpegLsCodec.cs

```diff
@@ -252,65 +253,82 @@ public override unsafe void Encode(DicomPixelData oldPixelData, DicomPixelData n
-                    }
+                        IByteBuffer frameData = oldPixelData.GetFrame(frame);
 
-                    PinnedByteArray frameArray = new PinnedByteArray(frameData.Data);
+                        //Converting photmetricinterpretation YbrFull or YbrFull422 to RGB
+                        if (oldPixelData.PhotometricInterpretation == PhotometricInterpretation.YbrFull)
+                        {
+                            frameData = PixelDataConverter.YbrFullToRgb(frameData);
+                        }
+                        else if (oldPixelData.PhotometricInterpretation == PhotometricInterpretation.YbrFull422)
+                        {
+                            frameData = PixelDataConverter.YbrFull422ToRgb(frameData, oldPixelData.Width);
+                        }
 
-                    byte[] jpegData = new byte[frameData.Size];
+                        PinnedByteArray frameArray = new PinnedByteArray(frameData.Data);
 
-                    try
-                    {
-                        fixed (byte* jpegDataPointer = jpegData)
+                        jpegData = pool.Rent((int)frameData.Size);
+
+                        try
                         {
-                            uint jpegDataSize = 0;
-                            char[] errorMessage = new char[256];
+                            fixed (byte* jpegDataPointer = jpegData)
+                            {
+                                uint jpegDataSize = 0;
+                                char[] errorMessage = new char[256];
 
-                            CharlsApiResultType err = CharlsApiResultType.Unknown;
+                                CharlsApiResultType err = CharlsApiResultType.Unknown;
 
-                            if (Platform.Current.Equals(Platform.Type.win_x64))
-                                err = JpegLSEncode_winx64(jpegDataPointer, (uint)jpegData.Length, &jpegDataSize, (void*)frameArray.Pointer, (uint)frameArray.Count, ref jls, errorMessage);
-                            else
-                                err = JpegLSEncode(jpegDataPointer, (uint)jpegData.Length, &jpegDataSize, (void*)frameArray.Pointer, (uint)frameArray.Count, ref jls, errorMessage);
+                                if (Platform.Current.Equals(Platform.Type.win_x64))
+                                    err = JpegLSEncode_winx64(jpegDataPointer, (uint)jpegData.Length, &jpegDataSize, (void*)frameArray.Pointer, (uint)frameArray.Count, ref jls, errorMessage);
+                                else
+                                    err = JpegLSEncode(jpegDataPointer, (uint)jpegData.Length, &jpegDataSize, (void*)frameArray.Pointer, (uint)frameArray.Count, ref jls, errorMessage);
 
-                            Array.Resize(ref jpegData, (int)jpegDataSize);
+                                newJpegData = pool.Rent((int)jpegDataSize);
+                                Array.Copy(jpegData, newJpegData, newJpegData.Length);
```

Comment:
Have to use Array.Copy since Array.Resize creates new array of the specified size and copies the element.